engine/baml-lib/baml-core/src/generate/generate_ts_client/expression.rs

```diff
@@ -31,9 +31,12 @@ impl ToTypeScript for Expression {
                 Identifier::Ref(r) => format!("\"{}\"", r.join(".")),
                 Identifier::Primitive(p) => p.to_ts(),
             },
-            Expression::String(val) => format!("\"{}\"", val.replace('"', "\\\"")),
+            Expression::String(val) => {
+                format!("\"{}\"", val.replace('"', "\\\"").replace('\n', "\\n"))
```

Comment:
shoudln't we do this for every possible escaped character? or i guess just \n are special because quoted strings can capture them? For clarity sake, i think we should do the same for "\t" as well

Reply:
leaving a todo to use escape_default here